examples/llm_inference/ios/InferenceExample/OnDeviceModel.swift

```diff
@@ -0,0 +1,110 @@
+// Copyright 2024 The MediaPipe Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+import MediaPipeTasksGenAI
+
+final class OnDeviceModel {
+
+  private var inference: LlmInference! = {
+    let path = Bundle.main.path(forResource: "model_cpu", ofType: "tflite")!
+    let llmOptions = LlmInference.Options(modelPath: path)
+    return LlmInference(options: llmOptions)
+  }()
+
+  // Temporary to remove extra replace characters left over in model output.
+  // Remove this function once no longer necessary.
+  private func cleanOutput(_ string: String) -> String {
+    let replaceCharacter: Character = "\u{FFFD}"
+    return string.reduce("") { partialResult, next in
+      guard let last = partialResult.last else {
+        if next == replaceCharacter {
+          return partialResult
+        }
+        return String(next)
+      }
+
+      let lastCharacterIsWhitespace = last.unicodeScalars.reduce(false, {
+        $0 || CharacterSet.whitespacesAndNewlines.contains($1)
+      })
+      if next == replaceCharacter {
+        if lastCharacterIsWhitespace {
+          return partialResult
+        }
+        return partialResult + " "
```

Comment:
I am surprised we have to do this. We don't add extra padding for the other models. Are we removing too much whitespace in the preprocessor?

Reply:
The whitespace here isn't extra padding, it's a replacement for the `U+FFFD` characters left in the model output (see screenshot)
![image](https://github.com/googlesamples/mediapipe/assets/4529021/69557d8b-ddd5-4620-b418-014dc3d07aac)

The reason they're replaced with whitespace is because the model seems to only insert `U+FFFD` in places that should be whitespace.

This whole method is written a little illegibly via `reduce`, but since it's designed to be deleted later I was very lazy about writing it nicely.

Reply:
This issue is resolved with the latest model, so I removed this bit of code.